src/basic/path-util.h

```diff
@@ -99,10 +99,24 @@ int path_strv_make_absolute_cwd(char **l);
 char** path_strv_resolve(char **l, const char *root);
 char** path_strv_resolve_uniq(char **l, const char *root);
 
-int find_executable_full(const char *name, const char *root, bool use_path_envvar, char **ret_filename, int *ret_fd);
+int find_executable_full(
+        const char *name,
+        const char *root,
+        char **exec_search_paths,
+        bool use_path_envvar,
+        char **ret_filename,
+        int *ret_fd);
 static inline int find_executable(const char *name, char **ret_filename) {
-        return find_executable_full(name, /* root= */ NULL, true, ret_filename, NULL);
+        return find_executable_full(name, /* root= */ NULL, NULL, true, ret_filename, NULL);
 }
+bool find_executable_from_path(
```

Comment:
Since this is a helper that shouldn't be used outside of `path-util.c` you can exclude it from here and declare `find_executable_from_path()` as static instead.